packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/RealmWebSocketTransport.kt

```diff
@@ -0,0 +1,123 @@
+package io.realm.kotlin.mongodb.internal
+
+import io.realm.kotlin.internal.interop.RealmWebsocketHandlerCallbackPointer
+import io.realm.kotlin.internal.interop.sync.CancellableTimer
+import io.realm.kotlin.internal.interop.sync.WebSocketClient
+import io.realm.kotlin.internal.interop.sync.WebSocketObserver
+import io.realm.kotlin.internal.interop.sync.WebSocketTransport
+import io.realm.kotlin.internal.interop.sync.WebsocketEngine
+import io.realm.kotlin.internal.util.CoroutineDispatcherFactory
+import io.realm.kotlin.internal.util.DispatcherHolder
+import kotlinx.atomicfu.AtomicRef
+import kotlinx.atomicfu.atomic
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+public class RealmWebSocketTransport(
+    public val timeoutMs: Long
+) : WebSocketTransport {
+    // we need a single thread dispatcher to act like an event loop
+    private val dispatcherHolder: DispatcherHolder = CoroutineDispatcherFactory.managed("RealmWebsocketTransport").create()
```

Comment:
I feel that it is wrong level of parallelism here. Seems weird that we have the options to selection multiplexing or not, but multiple websocket will be operation on the same thread. I don't know if this would have any actual effect though 🤔

Reply:
Having a dispatcher per websocket will only benefit if we have a full-duplex apps running in parallel which is a rare use case. Most use cases are the same app sharing the same transport, which is more inlined with multiplexing becoming the default...
We can revisit these approximations in case of a performance issue.